scripts/apps/authoring/authoring/directives/AuthoringDirective.ts

```diff
@@ -1212,6 +1212,15 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
                         $scope.schema = _schema;
                     });
             };
+
+            $scope.setCustomValue = (field, value) => {
+                const extra = Object.assign({}, $scope.item.extra);
+
+                extra[field._id] = value || null;
+                $scope.item.extra = extra;
+
+                $scope.autosave($scope.item, 200);
```

Comment:
```
$scope.autosave({
    ...$scope.item,
    extra: {
        ...$scope.item.extra,
        [field._id]: value || null,
    },
}, 200);
```

I thought you would use immutable update instead of moving the mutation :) Avoiding mutations when easy is beneficial even outside React. I don't mind the move though, it looks a bit nicer that components added via client API don't manipulate extra `directly`.